tests/tools/test_bluenaas_tool.py

```diff
@@ -17,39 +21,408 @@ async def test_arun(httpx_mock):
+                additional_kwargs={"refusal": None},
+                response_metadata={
+                    "token_usage": {
+                        "completion_tokens": 177,
+                        "prompt_tokens": 4819,
+                        "total_tokens": 4996,
+                        "completion_tokens_details": {"reasoning_tokens": 0},
+                        "prompt_tokens_details": {"cached_tokens": 4736},
+                    },
+                    "model_name": "gpt-4o-mini-2024-07-18",
+                    "system_fingerprint": "fp_e2bde53e6e",
+                    "finish_reason": "stop",
+                    "logprobs": None,
+                },
+                id="run-6e963cac-2392-483a-8a4c-0ac385130aba-0",
+                usage_metadata={
+                    "input_tokens": 4819,
+                    "output_tokens": 177,
+                    "total_tokens": 4996,
+                },
+            ),
+        ]
+    )
+
+    messages_2 = [
+        *messages,
+        HumanMessage(
+            content="I agree",
+            additional_kwargs={},
+            response_metadata={},
+            id="20bf996e-8f9f-4ad8-a607-913bf230280d",
+        ),
+        AIMessage(
+            content="",
+            additional_kwargs={
+                "tool_calls": [
+                    {
+                        "id": "call_6kYSBUM43aq1zCsf3CHUjSo2",
+                        "function": {
+                            "arguments": '{"me_model_id":"https://great_memodel.com}',
+                            "name": "bluenaas-tool",
+                        },
+                        "type": "function",
+                    }
+                ],
+                "refusal": None,
+            },
+            response_metadata={
+                "token_usage": {
+                    "completion_tokens": 287,
+                    "prompt_tokens": 5005,
+                    "total_tokens": 5292,
+                    "completion_tokens_details": {"reasoning_tokens": 0},
+                    "prompt_tokens_details": {"cached_tokens": 4864},
+                },
+                "model_name": "gpt-4o-mini-2024-07-18",
+                "system_fingerprint": "fp_e2bde53e6e",
+                "finish_reason": "tool_calls",
+                "logprobs": None,
+            },
+            id="run-bafc80a4-66fc-4509-a491-b2ea0d0ab505-0",
+            tool_calls=[
+                {
+                    "name": "bluenaas-tool",
+                    "args": {
+                        "me_model_id": "https://great_memodel.com",
+                    },
+                    "id": "call_6kYSBUM43aq1zCsf3CHUjSo2",
+                    "type": "tool_call",
+                }
+            ],
+            usage_metadata={
+                "input_tokens": 5005,
+                "output_tokens": 287,
+                "total_tokens": 5292,
+            },
+        ),
+    ]
+    # Case where we call bluenaas after validating. Run simu
```

Comment:
Small typo

Reply:
Are you referring to the correct line ? I don't see it 😄